code/game/machinery/stasis_cage.dm

```diff
@@ -0,0 +1,307 @@
+/obj/machinery/stasis_cage
+	name = "stasis cage"
+	desc = "A high-tech animal cage, designed to keep contained fauna docile and safe."
+	icon = 'icons/obj/machinery/stasis_cage.dmi'
+	icon_state = "stasis_cage"
+	density = TRUE
+	layer = ABOVE_OBJ_LAYER
+	req_access = list(ACCESS_RESEARCH)
+	idle_power_usage = 0
+	active_power_usage = 5000
+	use_power = POWER_USE_IDLE
+
+	var/datum/wires/stasis_cage/wires
+	var/mob/living/contained
+	var/datum/gas_mixture/airtank
+	var/broken = FALSE
+	var/safety = TRUE
+
+	var/obj/item/cell/cell = null
+
+
+/obj/machinery/stasis_cage/Initialize()
+	. = ..()
+	airtank = new /datum/gas_mixture(src)
+	airtank.temperature = T20C
+	airtank.adjust_gas(GAS_OXYGEN, MOLES_O2STANDARD, FALSE)
+	airtank.adjust_gas(GAS_NITROGEN, MOLES_N2STANDARD)
+
+	cell = new /obj/item/cell(src)
+
+	var/mob/living/A = locate() in loc
+	if(!A)
+		release(A)
+	else
+		contain(A)
+
+	wires = new /datum/wires/stasis_cage(src)
+
+/obj/machinery/stasis_cage/Destroy()
+	release()
+	QDEL_NULL(airtank)
+	QDEL_NULL(contained)
+	QDEL_NULL(cell)
+	QDEL_NULL(wires)
+	return ..()
+
+/obj/machinery/stasis_cage/process()
```

Comment:
I think it does, if the subsystem is ticking slower it can be that the stasis goes back to 0 before it's processed again, since they run on different subsystems (one is mob life, the other is processing)